### PR TITLE
Environment variable decryption

### LINKED ISSUE #1: KMS integration
KMS integration is breadcrumbed around:
https://github.com/thepwagner/flotilla/blob/master/src/flotilla/client/db.py#L46
https://github.com/thepwagner/flotilla/blob/master/src/flotilla/agent/db.py#L57
The latter is also where .service files are loaded back from the DB: could start with hash validation (or add signatures?), then continue on to KMS from there...

### PR DESCRIPTION
**Pull request by thepwagner**
Fixes #1

### CODE REVIEW

File: src/flotilla/agent/db.py [hunk 1]
```diff
@@ -23,13 +24,14 @@ class FlotillaAgentDynamo(object):
     """
 
     def __init__(self, instance_id, service_name, status_table,
-                 assignments_table, revisions_table, units_table):
+                 assignments_table, revisions_table, units_table, kms):
```
Comment (DeviaVir) [inline src/flotilla/agent/db.py]: Shouldn't https://github.com/thepwagner/flotilla/blob/master/src/agent.py#L43 also be init'd with KMS now, like https://github.com/pebble/flotilla/pull/10/files#diff-8d6ef5887842a4613dc91cfeefc9f28bR24

Comment (thepwagner) [inline src/flotilla/agent/db.py]: Doh. yeah that's going to break stuff.

I'll take #1 and integration test this over xmas break (catching this --^ and probably some IAM fixes too: no permissions granted against the key atm).

**Comment by thepwagner** [comment]
@DeviaVir IAM rolled in +tested, good to go

**Comment by DeviaVir** [comment]
:rocket: awesome stuff (do we want to rebase before merging? lots of commits)

**Comment by thepwagner** [comment]
Waiting until after break to merge ugly tree; justifying laziness!

**Pull request merged**